flink-runtime/src/main/java/org/apache/flink/runtime/leaderelection/DefaultLeaderElectionService.java

```diff
@@ -503,43 +516,53 @@ private void runInLeaderEventThread(Runnable callback) {
 
     private void forwardErrorToLeaderContender(Throwable t) {
         synchronized (lock) {
-            if (leaderContender == null) {
+            if (leaderContenderRegistry.isEmpty()) {
                 fallbackErrorHandler.onFatalError(t);
                 return;
             }
 
-            if (t instanceof LeaderElectionException) {
-                leaderContender.handleError((LeaderElectionException) t);
-            } else {
-                leaderContender.handleError(new LeaderElectionException(t));
-            }
+            leaderContenderRegistry
+                    .values()
+                    .forEach(
+                            leaderContender -> {
+                                if (leaderContender == null) {
```

Comment:
:thinking: When can the value of the registry map be null? Even if that were possible (which I dont think it is), then we'd logging this message once per contender.

Reply:
:facepalm: